### DISCUSSION TITLE
Chrome dies after connection to site with cert on HSM

### DISCUSSION THREAD

**Discussion by msetina** [OpenSC/OpenSC#3207] [Q&A]
I am at a los finding good information why Chrome dies after connecting to site using a ceritifcate on a card. I am using 0.25.1 on Ubuntu 22.04.
Looking at a log I can only find some info about ATI discrete card having trouble called from Chrome. It would indicate it is not an OpenSC problem, but the closing occurs only when using a cert on a card. There is some time between connection to the site and the closing.

**Comment by frankmorgner** [comment]
Sorry, I can't follow you. Please be more detailed about what you're trying to do with what hardware/software and what actually fails.

**Reply by msetina** [reply]
As written I am using Chrome on Ubuntu 22.04 which is the only one without snap f-ing the access to the Smartcard, but I keep on loosing the browser window (crash without error message). It happens a couple of seconds after authenticating to a site with a smartcard containg the certificate. The smartcard is OpenPGP 3.4. 
The Chrome sends the crash report, but no newer version has fixed the problem. To comlicate things I see the same behavior on Debian 10 with Chrome. On Debian I also have Chromium and Firefox and none of them have the same problem. 
On Ubuntu I do not have the same luck as both those come as snap install.
My main machines are Ubuntu 22.04, so this is a problem which I have trouble locating.

**Comment by dengert** [comment]
FireFox esr works on Ubuntu 22.04 and Ubuntu 24.04 
https://www.mozilla.org/en-US/firefox/115.2.0/releasenotes/ 
leads 
https://www.mozilla.org/en-US/firefox/all/#product-desktop-esr
where you can pick the OS, and version
I installed it as firefox-esr on 22.04 and as a ~/firefox/firefox on 24.04 to test OpenSC

**Comment by dengert** [comment]
Also there has been some progress. See:
https://bugs.launchpad.net/ubuntu/+source/firefox/+bug/1967632
https://bugzilla.mozilla.org/show_bug.cgi?id=1734371

**Reply by msetina** [reply]
Thank you @dengert  for your work on this snap problem. It reminds me of a problem I had with python CSV and some MS warehousing which was rolling for 4 years, but at the end was merged and released. But I digress....
I am at a point where pure Debian does not look so bad to switch to.
This was started for Chrome problem, that is not fixed. Now I am hoping that maybe I can get FF or Chromium working with workarounds. For me it sould be acceptable, but normal user would just bail.
WIll report if I could make something on 22.04. Otherwise I will try upgrade to 24.04 and check there. If all fails I am going bookworm and Chromium or FF. This snap thing is half baked. 
I do not know what the Chrome problem is. I have not had a crash like that since Windows95 :-)

**Comment by dengert** [comment]
Sounds like a chrome bug, but to help debug if it is OpenSC problem, you can turn on OpenSC debugging and to see PKCS11 calls and responses, use opensc-spy. See: https://github.com/OpenSC/OpenSC/wiki/Using-OpenSC

**Reply by msetina** [reply]
I would be surprised if it is linked to PKCS11, but I can try, since Chromium does not show the problem. If I understand correctly there is also NSS inbetween.

**Reply by dengert** [reply]
Yes. NSS loads the PKCS11module.

**Reply by msetina** [reply]
My thinking is that somehow it is not trivial to separate on which level the problem occurs. If I compare PKCS11 communication between Chrome and Chromium and they are the same, then all that remains is Chrome to NSS.
Can we rule out some sort of memory leak messing up the communication?

**Comment by msetina** [comment]
I am inclined to say this is similar to #3136

**Comment by wisdom-in-snow** [comment]
This looks like the bug discussed in https://issues.chromium.org/issues/41482445 and fixed in OpenSC 0.26.1.

**Reply by msetina** [reply]
Will check, thank you.
